stories/translator/doc-elements.tsx

```diff
@@ -0,0 +1,115 @@
+    return (
+        <SchemaViewWrapper>
+            <Text.H2>Generate Schema</Text.H2>
+            <SchemaPreview>
+                <pre>{JSON.stringify(data, null, 2)}</pre>
+            </SchemaPreview>
+        </SchemaViewWrapper>
+    );
+};
+
+export const DocElement = () => {
+    // =========================================================================
+    // CONST, STATE, REFS
+    // =========================================================================
+    const formBuilderRef = useRef<IFormBuilderMethods>(null);
+    const [pageMode, setPageMode] = useState<string>("form-builder-mode");
+    const [schema, setSchema] = useState<IFrontendEngineData | null>(null);
+
+    // =========================================================================
+    // EVENT HANDLERS
+    // =========================================================================
+    const handleFormBuilderButton = (event) => {
+        event.preventDefault();
+        setPageMode("form-builder-mode");
+    };
+
+    const handleFormPreviewButton = (event) => {
+        event.preventDefault();
+        setPageMode("preview-mode");
+    };
+
+    const handleSchemaViewButton = (event) => {
+        event.preventDefault();
+        setPageMode("schema-mode");
+    };
+
+    // =========================================================================
+    // EFFECTS
+    // =========================================================================
+    useEffect(() => {
+        if (formBuilderRef.current) {
+            const generatedSchema = formBuilderRef.current?.generate();
+            setSchema(generatedSchema);
+        }
+    }, [formBuilderRef.current, pageMode]);
+
+    // =========================================================================
+    // RENDER FUNCTIONS
+    // =========================================================================
+
+    return (
+        <>
+            <IconWrapper>
+                <IconButton
+                    $active={pageMode === "form-builder-mode"}
+                    onClick={handleFormBuilderButton}
+                >
+                    <PencilIcon />
+                </IconButton>
+                <IconButton
+                    $active={pageMode === "preview-mode"}
+                    onClick={handleFormPreviewButton}
+                >
+                    <EyeIcon />
+                </IconButton>
+                <IconButton
+                    $active={pageMode === "schema-mode"}
+                    onClick={handleSchemaViewButton}
+                >
+                    <TabletIcon />
+                </IconButton>
+            </IconWrapper>
+            <ContentWrapper visible={pageMode === "form-builder-mode"}>
+                <FormBuilder ref={formBuilderRef} offset={5.1} />
+            </ContentWrapper>
+            <ContentWrapper visible={pageMode === "preview-mode"}>
+                <FormPreview data={schema} />
+            </ContentWrapper>
```

Comment:
can we conditionally render the form? currently any changes / error states are retained despite switching back to the form builder and making further changes - they should be cleared